File: readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);

int count_words(string text);

int count_sentences(string text);

int main(void)
{
    //Asks the user for input
    //Then calls the functions to get the number of letters, words and sentences in the text.
    string text = get_string("Text: ");
    int numOfLetters = count_letters(text);
    int numOfWords = count_words(text);
    int numOfSentences = count_sentences(text);
    //Coleman-Liau Index formula
    int L = (numOfLetters / (double) numOfWords) * 100;
    int S = (numOfSentences / (double) numOfWords) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    double grade = round(index);
    //Outputs grade
    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", (int) grade);
    }
}

int count_letters(string text)
{
    int letters = 0;
    int i;
    //Loops through every character in the text
    for (i = 0; i < strlen(text); i++)
    {
        //Checks if the char is in the alphabet to know that its a letter
        if (isalpha(text[i]))
        {
            letters += 1;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 1;
    int i;
    for (i = 0; i < strlen(text); i++)
    {
        //Checks if the char is an empty space to know that its the end of a word
        if (isspace(text[i]))
        {
            words += 1;
        }
    }
    return words;
}

int count_sentences(string text)
{
    int sentences = 0;
    int i;
    for (i = 0; i < strlen(text); i++)
    {
        //Checks if the char is a ! ? or . to indicate the end of a sentence
        if (text[i] == '!' || text[i] == '?' || text[i] == '.')
        {
            sentences += 1;
        }
    }
    return sentences;
}

File: sentimental-readability/readability.py
# TODO
from cs50 import get_string

# Takes users input using get_string from the cs50 python library
text = get_string('Text:   ')


def count_letters(text):
    letters = 0
    for i in range(len(text)):
        if text[i].isalnum() == True:
            letters += 1
    return letters


def count_words(text):
    words = 1
    for i in range(len(text)):
        if text[i] == ' ':
            words += 1
    return words


def count_sentences(text):
    sentences = 0
    for i in range(len(text)):
        if text[i] == '!' or text[i] == '?' or text[i] == '.':
            sentences += 1
    return sentences


# Calculating the number of letters and words and sentences by calling the functions that are defined above
numOfLetters = count_letters(text)
numOfWords = count_words(text)
numOfSentences = count_sentences(text)
# Coleman-Liau Index Formula
L = (numOfLetters / numOfWords) * 100
S = (numOfSentences / numOfWords) * 100
index = 0.0588 * L - 0.296 * S - 15.8
grade = round(index)

# Grade outputting
if grade >= 16:
    print('Grade 16+\n')
elif grade < 1:
    print('Before Grade 1\n')
else:
    print('Grade', grade, '\n')